File: python-package/geobr/read_meso_region.py
from geobr.utils import read_geobr_v2


def read_meso_region(
    year: int,
    code_meso: str = "all",
    simplified: bool = True,
    verbose: bool = False,
    output: str = "gpd",
    show_progress: bool = True,
    cache: bool = True,
):
    """Download spatial data of meso region as sf objects.

     Data at scale 1:250,000, using Geodetic reference system "SIRGAS2000" and CRS(4674)

   Parameters
    ----------
    year : int
        Year of the data.
    code_meso : str or int
        4-digit meso region code, state abbrev, two-digit code, or ``"all"``.
    simplified, verbose, output, show_progress, cache
        Standard geobr options.
    """

    return read_geobr_v2(
        "mesoregions",
        year,
        code=code_meso,
        simplified=simplified,
        output=output,
        show_progress=show_progress,
        cache=cache,
        verbose=verbose,
    )

File: r-package/R/read_meso_region.R
#' Download spatial data of meso regions
#'
#' @description
#' Data at scale 1:250,000, using Geodetic reference system "SIRGAS2000" and CRS(4674)
#'
#' @template year
#' @param code_meso The 4-digit code of a meso region. If the two-digit code or
#'        a two-letter uppercase abbreviation of a state is passed, (e.g. 33 or
#'        "RJ") the function will load all meso regions of that state. If
#'        `code_meso="all"` (Default), the function downloads all meso
#'        regions of the country.
#' @template simplified
#' @template as_sf
#' @template showProgress
#' @template cache
#' @template verbose
#'
#' @return An `"sf" "data.frame"` OR an `ArrowObject`
#'
#' @export
#' @family area functions
#'
#' @examplesIf identical(tolower(Sys.getenv("NOT_CRAN")), "true")
#' # Read specific meso region at a given year
#' meso <- read_meso_region(code_meso=3301, year = 2018)
#'
#' # Read all meso regions of a state at a given year
#' meso <- read_meso_region(code_meso="AM", year = 2018)
#' meso <- read_meso_region(code_meso=12, year = 2018)
#'
#' # Read all meso regions of the country at a given year
#' meso <- read_meso_region(code_meso="all", year = 2018)
#'
read_meso_region <- function(year,
                             code_meso = "all",
                             simplified = TRUE,
                             as_sf = TRUE,
                             showProgress = TRUE,
                             cache = TRUE,
                             verbose = TRUE){

  # Get metadata with data url addresses
  temp_meta <- select_metadata(
    geography="mesoregions",
    year = year,
    simplified = simplified,
    verbose = verbose
  )

  # check if metadata download failed
  if (is.null(temp_meta)) { return(invisible(NULL)) }

  # download files
  temp_arrw <- download_parquet(
    filename_to_download = temp_meta$file_name,
    showProgress,
    cache
  )

  # check if download failed
  if (is.null(temp_arrw)) { return(invisible(NULL)) }

  # FILTER
  temp_arrw <- filter_arrw(temp_arrw, code = code_meso)

  # convert to sf
  if(isTRUE(as_sf)){
    temp_arrw <- sf::st_as_sf(temp_arrw)
  }

  return(temp_arrw)

}
